packages/compass-collection/src/modules/tabs.ts

```diff
@@ -213,8 +213,11 @@ const createNewTab = (
       localAppRegistry,
       ...collectionMetadata,
     });
-    const component = React.createElement(collectionTabRole.component, {
-      store,
+    const component = React.createElement(AppRegistryProvider, {
+      localAppRegistry: localAppRegistry,
```

Comment:
Yep, that looks good! One thing here still needs handling, as we discussed in the tech design, this is finally a case for us to use `deactivateOnUnmount` prop and disable it, otherwise switching to some other workspace, like instance for example, cleans up `localAppRegistry` for the tab